fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushPredicateThroughJoin.java

```diff
@@ -114,22 +113,12 @@ public Rule build() {
 
     private Plan pushDownPredicate(LogicalJoin<GroupPlan, GroupPlan> joinPlan,
             List<Expression> joinConditions, List<Expression> leftPredicates, List<Expression> rightPredicates) {
-
-        Expression left = ExpressionUtils.and(leftPredicates);
-        Expression right = ExpressionUtils.and(rightPredicates);
-        //todo expr should optimize again using expr rewrite
-        Plan leftPlan = joinPlan.left();
-        Plan rightPlan = joinPlan.right();
-        if (!left.equals(BooleanLiteral.TRUE)) {
-            leftPlan = new LogicalFilter<>(left, leftPlan);
-        }
-
-        if (!right.equals(BooleanLiteral.TRUE)) {
-            rightPlan = new LogicalFilter<>(right, rightPlan);
-        }
+        // todo expr should optimize again using expr rewrite
+        Plan leftPlan = PlanUtils.filterOrSelf(leftPredicates, joinPlan.left());
+        Plan rightPlan = PlanUtils.filterOrSelf(rightPredicates, joinPlan.right());
```

Comment:
These conditions are lost
  
 if (!left.equals(BooleanLiteral.TRUE)) {
            leftPlan = new LogicalFilter<>(left, leftPlan);
        }

        if (!right.equals(BooleanLiteral.TRUE)) {
            rightPlan = new LogicalFilter<>(right, rightPlan);
        }

Reply:
Not lost.
Because I use `optional` to replace the `true`.

Reply:
When `BooleanLiteral.TRUE`, it means that `leftPredicates` is empty, so we will return self.